website/docs/script/sudt-script.mdx

```diff
@@ -0,0 +1,383 @@
+
+2. **Input and Output Amount Validation**: Next, the script collects the total amounts of UDTs from both input and output cells. It ensures that the total amount of UDTs being sent (outputs) does not exceed the amount being received (inputs). This is crucial for maintaining the integrity of the token transfers.
+
+Here’s a snippet of the code that illustrates these checks:
+
+```rust
+pub fn program_entry() -> i8 {
+    ckb_std::debug!("This is a sample UDT contract!");
+
+    let script = load_script().unwrap();
+    let args: Bytes = script.args().unpack();
+    ckb_std::debug!("script args is {:?}", args);
+
+    // Check if the script is in owner mode
+    if check_owner_mode(&args) {
+        return 0; // Success if in owner mode
+    }
+
+    // Collect amounts from input and output cells
+    let inputs_amount: u128 = match collect_inputs_amount() {
+        Ok(amount) => amount,
+        Err(err) => return err as i8,
+    };
+    let outputs_amount = match collect_outputs_amount() {
+        Ok(amount) => amount,
+        Err(err) => return err as i8,
+    };
+
+    // Validate that inputs are greater than or equal to outputs
+    if inputs_amount < outputs_amount {
+        return Error::InvalidAmount as i8; // Error if invalid amount
+    }
+
+    0 // Success
+}
+```
+
+3. **Error Handling**: The script also includes robust error handling. It defines a custom `Error` enum to manage various error conditions, such as when the input amount is invalid or when there are issues with encoding. This helps ensure that any problems are clearly communicated.
+
+This implementation ensures that UDT transactions are validated correctly, maintaining the integrity of token transfers on the CKB blockchain. By checking both the owner mode and the amounts, the script helps prevent errors and ensures smooth operation.
+
+### Collecting UDT Amount
+
+In the sudt Script, the functions `collect_inputs_amount` and `collect_outputs_amount` play a crucial role in gathering the total amounts of UDTs from the respective input and output cells.
+
+Here’s how they work:
+
+Both functions iterate through the relevant cells and collect the `cell_data` to calculate the total UDT amount. This is done using the following syscalls:
+
+- **`Source::GroupInput`**: This ensures that only the cells with the same script as the current running script are iterating
+- **`Source::GroupOutput`**: Similarly, this syscall only the cells with the same script as the current running script are iterating
+
+By using these specific [sources](/docs/script/syscalls-for-script#source), the script avoids issues related to different UDT cell types, ensuring that only the appropriate cells are processed. This is particularly important in a blockchain environment where multiple UDTs may exist, and we want to ensure that the calculations are accurate and relevant to the current UDT Script.
```

Comment:
By using these specific [sources](/docs/script/syscalls-for-script#source), the Script avoids issues related to different UDT Cell types, ensuring that only the appropriate Cells can be processed. This is particularly important in a blockchain environment where multiple UDTs may exist, to ensure accurate iteration of the relevant UDT Script.